### PR TITLE
feat: improve workspace api

### PR DESCRIPTION
**Pull request by speed2exe**
- deprecate the user of `AFWorkspace`, just use `Vec<AFWorkspace>`
- add option to include `member_count` for each workspace when fetching workspace

### CODE REVIEW

File: libs/client-api/src/http.rs [hunk 1]
```diff
@@ -627,16 +628,26 @@ impl Client {
     AppResponse::<()>::from_response(resp).await?.into_error()
   }
 
+  pub async fn get_workspaces(&self) -> Result<Vec<AFWorkspace>, AppResponseError> {
+    self
+      .get_workspaces_opt(QueryWorkspaceParam::default())
+      .await
+  }
+
   #[instrument(level = "info", skip_all, err)]
```
Comment (speed2exe) [inline libs/client-api/src/http.rs:637]: @qinluhe use for additional member count field, pass in `QueryWorkspaceParam` with `include_member_count` set to true

**Comment by qinluhe** [comment]
LGTM

**Review by qinluhe — APPROVED** [review]
LGTM

File: libs/client-api/src/http.rs [hunk 2]
```diff
@@ -627,16 +628,26 @@ impl Client {
...
-  pub async fn get_workspaces(&self) -> Result<AFWorkspaces, AppResponseError> {
```
Comment (appflowy) [inline libs/client-api/src/http.rs:631]: Is it compatible with old api endpoint?

Comment (speed2exe) [reply libs/client-api/src/http.rs:631]: yes it is backward compatible. frontend appflowy have to change from using `AFWorkspaces` to `Vec<Workspace>` when update, but that's just rust structure, the http endpoint behaviour is same for old requests

**Review by lilpoozie2005 — APPROVED** [review]

**Review by lilpoozie2005 — APPROVED** [review]

**Review by lilpoozie2005 — APPROVED** [review]

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**